src/utils.ml

```diff
@@ -152,7 +167,11 @@ module Ast_io = struct
   let read input_source ~input_kind =
     try
       match input_source with
-      | Stdin -> from_channel stdin ~input_kind
+      | Stdin ->
+          (match input_kind with
+          | Necessarily_binary -> set_binary_mode_in stdin true
+          | _ -> ());
```

Comment:
`input_kind` can be either `Necessarily_binary` or `Possibly_source`. I think binary mode should be set on both cases, because both cases may end up reading binary data.

Reply:
Changed and tested. (The "Second" problem still remains open)